### PR TITLE
[ERE-1746] [ERE-1747] Report Designer usability fixes

### PR DESCRIPTION
**Pull request by ppettitau**
* ERE-1746 Add helpful text to Access Groups field to make it clear what setting the value does
* ERE-1747 Refactor report filters to make them conditionally display depending if the user has chosen to add them to the report via checkbox

### CODE REVIEW

**Comment by ppettitau** [comment]
image
image
image

File: rdrf/report/templates/report_designer.html [hunk 1]
```diff
@@ -209,37 +225,50 @@ <h1>
...
-                </ul>
-            </div>
-            <div class="offset-sm-2 col-sm-10 form-text">
-                {% trans "Filter the report to only include patients that have consented to the selected consent items." %}
-            </div>
-        </div>
+        <legend>Report Filters</legend>
+        <ul class="list-group offset-sm-2 col-sm-10">
+            <li class="list-group-item">
+                {{ form.filter_by_working_groups|add_attr:"class,form-check-input report-filter-toggle" }}
```
Comment (ScriptSmith) [inline rdrf/report/templates/report_designer.html:231]: Didn't know you could do that :+1:

**Review by ScriptSmith — APPROVED** [review]

File: rdrf/report/templates/report_designer.html [hunk 2]
```diff
@@ -209,37 +225,50 @@ <h1>
...
+                <label for="id_filter_by_working_groups">{{ form.filter_by_working_groups.label }}</label>
+                <div class="row d-none report-filter">
+                    <label for="id_filter_working_groups" class="visually-hidden">{{ form.filter_working_groups.label }}</label>
+                    <div class="mt-2">
+                        {{ form.filter_working_groups }}
+                    </div>
+                    <div class="form-text">
+                        {% trans "Filter the report to only include patients within the selected working groups." %}
```
Comment (ScriptSmith) [inline rdrf/report/templates/report_designer.html]: Maybe this should say:

> Filter the report to only include patients within **any of** the selected working groups.

And the other should say:

> Filter the report to only include patients that have consented to **all of** the selected consent items.

(without the bold)

**Review by ScriptSmith — CHANGES_REQUESTED** [review]

File: rdrf/report/templates/report_designer.html (hunk 2 above)
Comment (ppettitau) [reply rdrf/report/templates/report_designer.html]: Good point.

Comment (ppettitau) [reply rdrf/report/templates/report_designer.html]: Updated.

**Review by ScriptSmith — APPROVED** [review]

**Pull request merged**